src/bytes.js

```diff
@@ -0,0 +1,1443 @@
+const supportedEncodings = {
+    "utf": "utf-8",
+    "utf8": "utf-8",
+    "utf-8": "utf-8",
+    "ascii": "ascii"
+};
+
+function normalizeEncoding (encoding) {
+    const normalized = encoding.replace(/\s+/g, "").toLowerCase();
+    const supported = supportedEncodings[normalized];
+    if (supported === undefined) {
+        return encoding;
+    } else {
+        return supported;
+    }
+}
+
+/**
+ * @constructor
+ * @param {*} source
+ * @param {Sk.builtin.str=} encoding
+ * @param {Sk.builtin.str=} errors
+ * @return {Sk.builtin.bytes}
+ */
+
+Sk.builtin.bytes = function (source, encoding, errors) {
+    let view;
+    let arr;
+
+    if (!(this instanceof Sk.builtin.bytes)) {
+        return new Sk.builtin.bytes(...arguments);
+    }
+    Sk.builtin.pyCheckArgsLen("bytes", arguments.length, 0, 3);
+
+    if (errors === undefined) {
+        errors = "strict";
+    } else if (!(errors instanceof Sk.builtin.str)) {
+        throw new Sk.builtin.TypeError("bytes() argument 2 must be str, not " + Sk.abstr.typeName(errors));
+    } else {
+        errors = errors.v;
+    }
+    if (encoding === undefined) {
+        encoding = Sk.builtin.str.$utf8;
+    }
+    if (!(errors === "strict" || errors === "ignore" || errors === "replace")) {
+        throw new Sk.builtin.NotImplementedError("'" + errors + "' error handling not implemented in Skulpt");
+    }
+    if (arguments.length === 0) {
+        return new Sk.builtin.bytes(0);
+    }
+    if (arguments.length === 1) {
+        if (Sk.builtin.checkInt(source)) {
+            source = Sk.builtin.asnum$(source);
+            arr = new Uint8Array(source);
+        } else if (source instanceof Sk.builtin.bytes) {
+            return source;
+        } else if (Array.isArray(source)) {
+            // Internal fast path
+            Sk.asserts.assert(source.every((x) => (x >= 0) && (x < 256)),
+                              "Bad internal call to bytes with array object");
+            arr = new Uint8Array(source);
+        } else if (source instanceof Uint8Array) {
+            // Internal fast path
+            arr = source;
+        } else if ((Sk.builtin.checkIterable(source) && !(source instanceof Sk.builtin.str))) {
+            const final = [];
+            for (let iter = Sk.abstr.iter(source), item = iter.tp$iternext();
+                item !== undefined;
+                item = iter.tp$iternext()) {
+                if (Sk.builtin.checkInt(item)) {
+                    item = Sk.builtin.asnum$(item);
+                    if (item >= 0 && item <= 256) {
+                        final.push(item);
+                    } else {
+                        throw new Sk.builtin.ValueError("bytes must be in range(0, 256)");
+                    }
+                } else {
+                    throw new Sk.builtin.TypeError("'" + Sk.abstr.typeName(item) + "' " + "object cannot be interpreted as an integer");
+                }
+            }
+            arr = new Uint8Array(final);
```

Comment:
how careful do we need to be here
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array/Uint8Array#Browser_compatibility
`new Uint8Array(iterable)` support